web/src/features/column-visibility/hooks/useColumnVisibility.ts

```diff
@@ -4,12 +4,51 @@ import useLocalStorage from "@/src/components/useLocalStorage";
 import { useEffect } from "react";
 import { isEqual } from "lodash";
 
+const readStoredVisibilityState = (
+  localStorageKey: string,
+): VisibilityState => {
+  if (typeof window === "undefined") {
+    return {};
+  }
+  try {
+    const storedValue = localStorage.getItem(localStorageKey);
+    return storedValue ? JSON.parse(storedValue) : {};
+  } catch (error) {
+    console.error("Error reading from local storage", error);
+    return {};
+  }
+};
+
 function useColumnVisibility<TData>(
   localStorageKey: string,
   columns: LangfuseColumnDef<TData>[],
 ) {
   const initialVisibilityState = () => {
     const visibilityState: VisibilityState = {};
+    const storedVisibilityState = readStoredVisibilityState(localStorageKey);
+
+    // With virtual detail columns we must ensure state in local storage is valid given current project data and possibly upsert
```

Comment:
why is there a need to monitor this?

we could we not keep localstorage as is, ignore non-existing columns on read, and write valid set of columns on change